### PR TITLE
refactor(SecurityHub): create class

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context

We want to move the outputs to a decoupled component.

**https://github.com/prowler-cloud/prowler/pull/4127 must be reviewed first.**

### Description

Create the SecurityHub class and update code accordingly.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/providers/aws/utils.py [hunk 1]
```diff
@@ -153,16 +153,27 @@ def set_mocked_aws_provider(
     return provider
 
 
-def set_default_provider_arguments(arguments: Namespace) -> Namespace:
+def set_default_provider_arguments(input_arguments: Namespace) -> Namespace:
+
+    arguments = Namespace
     arguments.status = []
+    if hasattr(input_arguments, "status") and input_arguments.status:
```
Comment (pedrooot) [inline tests/providers/aws/utils.py]:
```suggestion
    if getattr(input_arguments, 'status', None):
```

File: tests/providers/aws/utils.py [hunk 2]
```diff
@@ -153,16 +153,27 @@ def set_mocked_aws_provider(
...
     arguments.verbose = False
     arguments.only_logs = False
     arguments.unix_timestamp = False
     arguments.shodan = None
     arguments.security_hub = False
+
     arguments.send_sh_only_fails = False
+    if (
+        hasattr(input_arguments, "send_sh_only_fails")
+        and input_arguments.send_sh_only_fails
```
Comment (pedrooot) [inline tests/providers/aws/utils.py]:
```suggestion
        getattr(input_arguments, 'send_sh_only_fails', None)
```

**Review by pedrooot — COMMENTED** [review]
Please, review my comments. Thanks!

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            # Iterate findings by region
+            for region, findings in security_hub_findings_per_region.items():
+                # Send findings to Security Hub
+                logger.info(f"Sending findings to Security Hub in the region {region}")
+
+                security_hub_client = self._session.client(
+                    "securityhub", region_name=region
+                )
+
+                success_count = self.__send_findings_to_security_hub__(
```
Comment (jfagoagas) [inline prowler/lib/outputs/security_hub/security_hub.py]:
```suggestion
                success_count += self.__send_findings_to_security_hub__(
```

File: tests/providers/aws/utils.py (hunk 1 above)
Comment (jfagoagas) [reply tests/providers/aws/utils.py]: Great one!

File: tests/providers/aws/utils.py (hunk 2 above)
Comment (jfagoagas) [reply tests/providers/aws/utils.py]: Great one!

**Review by pedrooot — DISMISSED** [review]
LGTM!!

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        self._provider = provider
+        self._session = provider.session.current_session
+        self._account = provider.identity.account
+        self._partition = provider.identity.partition
+
+    @property
+    def findings_per_region(self):
+        return self._findings_per_region
+
+    def prepare_security_hub_findings(
```
Comment (MrCloudSec) [inline prowler/lib/outputs/security_hub/security_hub.py]:
```suggestion
    def prepare(
```

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+            finding_json_asff = fill_json_asff(self._provider, finding)
+
+            # Include that finding within their region in the JSON format
+            security_hub_findings_per_region[region].append(
+                finding_json_asff.dict(exclude_none=True)
+            )
+
+        return security_hub_findings_per_region
+
+    def verify_security_hub_integration_enabled_per_region(
```
Comment (MrCloudSec) [inline prowler/lib/outputs/security_hub/security_hub.py]:
```suggestion
    def verify(
```

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+                )
+
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__} -- [{error.__traceback__.tb_lineno}]:{error} in region {region}"
+            )
+        return success_count
+
+    # Move previous Security Hub check findings to ARCHIVED (as prowler didn't re-detect them)
+    def resolve_security_hub_previous_findings(
```
Comment (MrCloudSec) [inline prowler/lib/outputs/security_hub/security_hub.py]:
```suggestion
    def resolve(
```

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+                )
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__} -- [{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+        finally:
+            return prowler_integration_enabled
+
+    def batch_send_to_security_hub(
```
Comment (MrCloudSec) [inline prowler/lib/outputs/security_hub/security_hub.py]:
```suggestion
    def send(
```

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+                success_count += self.__send_findings_to_security_hub__(
+                    findings_to_archive, region, security_hub_client
+                )
+            except Exception as error:
+                logger.error(
+                    f"{error.__class__.__name__} -- [{error.__traceback__.tb_lineno}]:{error} in region {region}"
+                )
+        return success_count
+
+    def __send_findings_to_security_hub__(
```
Comment (MrCloudSec) [inline prowler/lib/outputs/security_hub/security_hub.py]:
```suggestion
    def __batch_import_findings__(
```

File: prowler/lib/outputs/security_hub/security_hub.py [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+            logger.error(
+                f"{error.__class__.__name__} -- [{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+        finally:
+            return prowler_integration_enabled
+
+    def batch_send_to_security_hub(
+        self,
+        security_hub_findings_per_region: dict,
```
Comment (MrCloudSec) [inline prowler/lib/outputs/security_hub/security_hub.py:126]: Why not getting the `security_hub_findings_per_region` argument of all the functions from the class attribute?

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Love it! Please, check my comments.

File: prowler/lib/outputs/security_hub/security_hub.py (hunk 9 above)
Comment (MrCloudSec) [reply prowler/lib/outputs/security_hub/security_hub.py:126]: We will do it in the future.

**Comment by jfagoagas** [comment]
Closed in favor of https://github.com/prowler-cloud/prowler/pull/4397.

**Pull request closed without merging**